packages/app-mobile/root.tsx

```diff
@@ -822,6 +826,60 @@ class AppComponent extends React.Component {
 		}
 	}
 
+	private async openCallbackUrl(url: string) {
+		if (!isCallbackUrl(url)) {
+			return;
+		}
+		reg.logger().info(`openUrl ${url}`);
+
+		const { command, params } = parseCallbackUrl(url);
+		switch (command) {
+		case CallbackUrlCommand.OpenNote:
+			if (!await Note.load(params.id)) {
+				reg.logger().error(`note ${params.id} doesn't exist`);
+				return;
+			}
+			await this.prepareToNavigate();
+			await this.props.dispatch({
+				type: 'NAV_GO',
+				noteId: params.id,
+				routeName: 'Note',
+			});
```

Comment:
Could you refactor to avoid the prepareToNavigate/dispatch repetition?

Reply:
Done